src/WebJobs.Script/OutOfProc/Http/DefaultHttpWorkerService.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Formatting;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.WebApiCompatShim;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Azure.WebJobs.Script.Description;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Newtonsoft.Json;
+
+namespace Microsoft.Azure.WebJobs.Script.OutOfProc.Http
+{
+    public class DefaultHttpWorkerService : IHttpWorkerService
+    {
+        private readonly HttpClient _httpClient;
+        private readonly HttpWorkerOptions _httpWorkerOptions;
+        private readonly ILogger _logger;
+
+        public DefaultHttpWorkerService(HttpClient httpClient, IOptions<HttpWorkerOptions> httpWorkerOptions, ILoggerFactory loggerFactory)
+        {
+            _httpClient = httpClient ?? throw new ArgumentNullException(nameof(httpClient));
+            _httpWorkerOptions = httpWorkerOptions.Value;
+            _logger = loggerFactory.CreateLogger<DefaultHttpWorkerService>();
+        }
+
+        public Task InvokeAsync(ScriptInvocationContext scriptInvocationContext)
+        {
+            if (Utility.IsSimpleHttpTriggerFunction(scriptInvocationContext.FunctionMetadata))
```

Comment:
>Utility.IsSimpleHttpTriggerFunction(scriptInvocationContext.FunctionMetadata [](start = 16, length = 76)

Would it feel more natural to have something like scriptInvocationContext.FunctionMetadata.IsSimpleHttpTriggerFunction()?

Reply:
Agreed. Will move this to extensions methods